### ISSUE TITLE
Replace sequential greedy optimization with simultaneous joint optimization

### ISSUE THREAD

**Issue by Nuno-Monteiro** [GoCleverOrg/poc-homography#121]
## Summary
Replace the current sequential greedy optimization in `auto_calibrator.py` with simultaneous joint optimization of all 6 camera parameters (Δpan, Δtilt, Δroll, ΔX, ΔY, ΔZ). The current approach optimizes parameters one-at-a-time using `scipy.optimize.minimize_scalar`, which gets stuck in local minima because parameters are correlated. Simultaneous optimization will use multivariate methods like `scipy.optimize.least_squares` or `scipy.optimize.minimize` to solve all parameters together, enabling the optimizer to navigate parameter correlations and find better global solutions.

## Context
The current `AutoCalibrator` class implements a greedy sequential approach that optimizes parameters in fixed order: pan → tilt → height → GPS lat → GPS lon. Each parameter is optimized while holding all others constant using Brent's method (`minimize_scalar`), which is a univariate optimizer designed for single-parameter optimization.

This sequential approach has fundamental limitations:
- **Parameter correlations cause local minima**: Height vs tilt are weakly observable in single-frame scenarios—optimizing one affects the optimal value of the other. Similarly, pan offset vs ΔX/ΔY are coupled depending on camera geometry.
- **Order-dependent results**: Different optimization orders can yield different final parameters.
- **Suboptimal convergence**: Greedy optimization cannot explore the full parameter space efficiently.

The new reprojection error objective function (replacing ECC correlation) is better suited for multivariate optimization because:
- Reprojection error is differentiable with respect to all camera parameters
- It provides gradient information that multivariate optimizers can exploit
- It operates on feature correspondences rather than mask correlation, enabling more precise calibration

This issue addresses the optimization algorithm only. Multi-frame calibration (using multiple camera views) is deferred to future work.

### In Scope
- Replace `_optimize_parameter()` and sequential optimization loop with simultaneous joint optimization
- Implement multivariate optimization for all 6 parameters: Δpan, Δtilt, Δroll, ΔX, ΔY, ΔZ
- Use `scipy.optimize.least_squares` with 'lm' (Levenberg-Marquardt) or 'trf' (Trust Region Reflective) method
- Alternative: use `scipy.optimize.minimize` with 'L-BFGS-B' for bounded optimization
- Define parameter bounds as constraints for the optimizer
- Update objective function to accept parameter vector and return scalar error (or residual vector for `least_squares`)
- Maintain support for optional GPS-based position optimization (if GPS converter provided)
- Preserve existing `CalibrationResult` output format
- Update class docstring and module-level documentation to reflect simultaneous optimization
- Update convergence criteria to use optimizer defaults or define joint convergence tolerance

### Out of Scope
- Multi-frame calibration (using multiple camera images/positions)
- Implementing coarse-to-fine optimization strategies (grid search → fine optimization)
- Multiple random restart mechanisms to escape local minima
- Automatic differentiation or custom gradient implementations
- Changes to reprojection error objective function implementation
- Modifications to `CameraGeometry` class or homography computation
- Changes to `CalibrationResult` data structure beyond existing fields
- GUI or visualization of optimization progress
- Benchmarking or performance comparison with sequential approach

## Definition of Done
- [ ] `auto_calibrator.py` uses multivariate optimizer instead of `minimize_scalar`
- [ ] All 6 parameters optimized simultaneously in single optimization call
- [ ] Parameter bounds enforced via optimizer constraints
- [ ] Objective function accepts parameter vector (6-element array) and returns error metric
- [ ] GPS-based position optimization still works when `coordinate_converter` provided
- [ ] `CalibrationResult` output includes same fields as before (no breaking changes)
- [ ] Module-level docstring updated to describe simultaneous optimization approach
- [ ] Class docstring for `AutoCalibrator` updated with new optimization method
- [ ] Existing unit tests pass or are updated to reflect new optimization approach
- [ ] No references to sequential optimization or `_optimize_parameter()` method remain

### Multivariate Optimization Strategy

Replace the sequential parameter loop with a single multivariate optimization call.

**Context**: The current `run()` method iterates through parameters sequentially (lines 506-593 in `auto_calibrator.py`). This entire loop must be replaced with a joint optimization that solves for all 6 parameters simultaneously. Multivariate optimizers can handle parameter correlations that univariate optimizers cannot.

**Requirements**:
- Define parameter vector: `params = [Δpan, Δtilt, Δroll, ΔX, ΔY, ΔZ]` (6 elements)
- Use `scipy.optimize.least_squares` with method='trf' or method='lm'
- Alternative: `scipy.optimize.minimize` with method='L-BFGS-B' for bounded optimization
- Return either residual vector (for `least_squares`) or scalar error (for `minimize`)
- Remove `_optimize_parameter()` method entirely

**Reference**: `scipy.optimize.least_squares` documentation: https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.least_squares.html

### Parameter Vector and Bounds

Define parameter representation and constraints for the optimizer.

**Context**: Current implementation uses relative bounds per parameter (e.g., pan ±10°, tilt ±5°). These bounds must be translated into optimizer-compatible constraint format. Parameter vector must combine angular parameters (degrees) with spatial parameters (meters), requiring careful handling of units and scales.

**Requirements**:
- Parameter vector: `[Δpan_deg, Δtilt_deg, Δroll_deg, ΔX_m, ΔY_m, ΔZ_m]`
- Initial guess: `[0, 0, 0, 0, 0, 0]` (current parameters are the starting point)
- Bounds specified as `bounds=(lower, upper)` where each is 6-element array
- Lower bounds: `[-pan_bounds, -tilt_bounds, -roll_bounds, -xy_bounds, -xy_bounds, -height_bounds]`
- Upper bounds: `[+pan_bounds, +tilt_bounds, +roll_bounds, +xy_bounds, +xy_bounds, +height_bounds]`
- Handle absolute limits from `CameraGeometry` constants (e.g., `TILT_MIN`, `HEIGHT_MIN`)
- Apply bounds to prevent physically invalid configurations

**Reference**: Current bounds at lines 143-147 in `auto_calibrator.py`. Existing `CameraGeometry` validation at lines 548-560.

### Objective Function Restructuring

Adapt objective function to accept parameter vector instead of single scalar.

**Context**: Current `objective()` nested function (lines 427-465) operates on a single parameter value. The new objective must accept a 6-element parameter vector, apply those deltas to current camera parameters, regenerate homography, and compute error. For `least_squares`, return residual vector; for `minimize`, return scalar error.

**Requirements**:
- Function signature: `objective(params: np.ndarray) -> Union[float, np.ndarray]`
- Extract deltas from parameter vector: `Δpan, Δtilt, Δroll, ΔX, ΔY, ΔZ = params`
- Apply deltas to base camera parameters (stored at optimization start)
- Update `CameraGeometry` with perturbed parameters via `set_camera_parameters()`
- Compute reprojection error (or correlation if still using ECC temporarily)
- Return appropriate format: scalar for `minimize`, residual vector for `least_squares`

**Note**: If using reprojection error, residuals are per-feature-point errors. If using correlation (temporarily), convert to scalar loss: `loss = 1.0 - correlation`.

**Reference**: Existing objective function at lines 427-465 in `auto_calibrator.py`.

### GPS Position Integration

Maintain GPS-based position optimization when coordinate converter is available.

**Context**: Current implementation supports GPS lat/lon optimization when `camera_gps` and `coordinate_converter` are provided (lines 437-456). In simultaneous optimization, ΔX and ΔY parameters replace GPS lat/lon deltas, but the GPS converter must still be used to initialize the base position.

**Requirements**:
- If GPS enabled: use `coordinate_converter.gps_to_local_xy(lat, lon)` to initialize base world position
- Apply ΔX, ΔY deltas directly to world coordinates: `w_pos[0] += ΔX`, `w_pos[1] += ΔY`
- No need to optimize GPS lat/lon separately—ΔX/ΔY handle position directly
- Update final `optimized_params` dictionary to include GPS coordinates if converter available
- Convert final world position back to GPS: `lat, lon = converter.local_xy_to_gps(x, y)` (if inverse method exists)

**Reference**: GPS optimization logic at lines 437-456 and 577-591 in `auto_calibrator.py`.

### Convergence Criteria and Tolerances

Define stopping conditions for multivariate optimization.

**Context**: Current per-parameter tolerance is 0.001 with max 20 iterations per parameter (lines 150-151). For joint optimization, convergence criteria apply to the entire parameter vector. `scipy.optimize.least_squares` and `minimize` have default tolerances that may differ from current behavior.

**Requirements**:
- Use optimizer defaults initially: `ftol` (function tolerance), `xtol` (parameter tolerance), `gtol` (gradient tolerance)
- Optionally override: `ftol=1e-4`, `xtol=1e-4` for `least_squares`
- For `minimize`: `options={'ftol': 1e-4, 'maxiter': 100}`
- Remove per-parameter `maxiter` and `xatol` (no longer applicable)
- Global timeout still enforced: check `_check_timeout()` in objective function, raise exception to halt optimizer

**Reference**: Current convergence criteria at lines 472-475 in `auto_calibrator.py`. Optimizer documentation for default tolerances.

### Documentation Updates

Update module and class docstrings to reflect simultaneous optimization.

**Context**: Module-level docstring (lines 1-49) describes "sequential parameter optimization" and "greedy sequential approach." Class docstring (lines 96-141) explains "each parameter is optimized while holding others constant." These descriptions are obsolete after this change.

**Requirements**:
- Replace "sequential" with "simultaneous joint optimization"
- Update algorithm description: "Uses multivariate optimization (scipy.optimize.least_squares or minimize) to solve all 6 parameters together"
- Remove references to Brent's method and univariate optimization
- Update example usage to reflect new optimization behavior
- Specify optimizer method used: "Trust Region Reflective (trf)" or "Levenberg-Marquardt (lm)" or "L-BFGS-B"
- Clarify that parameters are correlated and joint optimization handles correlations
- Update convergence criteria documentation to reflect multivariate tolerances

**Reference**: Current docstrings at lines 1-49 (module) and 96-141 (class).

### CalibrationResult Compatibility

Ensure output format remains unchanged for downstream consumers.

**Context**: `CalibrationResult` dataclass (lines 66-93) is used by calling code to inspect optimization outcomes. Fields include `optimized_params`, `improvement`, `steps_completed`, etc. With joint optimization, there are no distinct "steps" per parameter.

**Requirements**:
- `steps_completed` field: set to 1 (single joint optimization step) or number of optimizer iterations
- `optimized_params` dictionary: maintain same keys (pan_deg, tilt_deg, height_m, w_pos, K, etc.)
- `timeout_reached`: still check global timeout, halt optimization if exceeded
- `improvement`: still compute as `final_error - initial_error` or `final_correlation - initial_correlation`
- `message`: update success/failure messages to reflect joint optimization
- No new fields required (no breaking changes to dataclass)

**Reference**: `CalibrationResult` dataclass at lines 66-93. Result creation at lines 632-644 in `auto_calibrator.py`.

### Current Sequential Optimization (Before)
```python
# Current approach: optimize parameters one-by-one
for i, (param_name, bounds_offset) in enumerate(param_sequence, start=1):
    optimal_value = self._optimize_parameter(
        param_name, current_value, bounds, i, total_steps
    )
    # Update parameter and regenerate homography
    params[param_name] = optimal_value
    self._set_parameters(params)
```

### Simultaneous Joint Optimization (After)
```python
# New approach: optimize all parameters together
initial_params = np.array([0, 0, 0, 0, 0, 0])  # Δpan, Δtilt, Δroll, ΔX, ΔY, ΔZ
bounds = (
    [-pan_bounds, -tilt_bounds, 0, -xy_bounds, -xy_bounds, -height_bounds],
    [+pan_bounds, +tilt_bounds, 0, +xy_bounds, +xy_bounds, +height_bounds]
)

result = least_squares(
    objective_function,
    initial_params,
    bounds=bounds,
    method='trf',
    ftol=1e-4,
    xtol=1e-4
)

optimized_params = result.x  # Final parameter vector
```

### Objective Function Signature (After)
```python
def objective(params: np.ndarray) -> np.ndarray:
    """
    Compute residuals for all feature points given parameter deltas.

    Args:
        params: [Δpan, Δtilt, Δroll, ΔX, ΔY, ΔZ]

    Returns:
        Residual vector (one element per feature point) for least_squares
        OR scalar error for minimize
    """
    Δpan, Δtilt, Δroll, ΔX, ΔY, ΔZ = params

    # Apply deltas to base parameters
    perturbed_params = base_params.copy()
    perturbed_params['pan_deg'] += Δpan
    perturbed_params['tilt_deg'] += Δtilt
    perturbed_params['w_pos'][0] += ΔX
    perturbed_params['w_pos'][1] += ΔY
    perturbed_params['w_pos'][2] += ΔZ

    # Update geometry and compute error
    self._set_parameters(perturbed_params)
    error = self._compute_reprojection_error()  # or correlation

    return error  # residual vector or scalar
```
